lib/msf/core/post/android/priv.rb

```diff
@@ -0,0 +1,35 @@
+# -*- coding: binary -*-
+
+require 'msf/core/post/common'
+require 'msf/core/post/file'
+require 'msf/core/post/unix'
+
+module Msf
+class Post
+module Android
+module Priv
+
+  include Msf::Post::Common
+
+  public
+
+  # Returns whether we are running as root or not.
+  #
+  # @return [Boolean] TrueClass if as root, otherwise FalseClass.
+  def is_root?
```

Comment:
Same idea, but not quite. The linux version uses id -u, but the android version doesn't understand -u, so I couldn't reuse the code.